File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int right = heights.length -1;
        int left =0;
        int maxArea =0;

        while(left < right){
            int currentArea =(right -left) * Math.min(heights[left], heights[right]);

            if (heights[left] < heights[right]){
                left++;
            }else{
                right--;
            }
            maxArea = Math.max(currentArea,maxArea);
        }
        return maxArea;
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left=0
        tot=0
        left=0
        minLen = 1000
        for right in range(len(nums)):
            tot+=nums[right]

            while tot >= target:
                tot-=nums[left]
                minLen = min(right-left+1,minLen)
                left+=1
                
        
        if minLen == 1000:
            return 0
        else:
            return minLen
